agent/start.go

```diff
@@ -35,7 +36,18 @@ func StartDaemon(ctx context.Context) (*Client, error) {
 	}
 
 	cmd := exec.Command(os.Args[0], "agent", "run", logFile)
-	cmd.Env = append(os.Environ(), "FLY_NO_UPDATE_CHECK=1")
+
+	env := os.Environ()
+	env = append(env, "FLY_NO_UPDATE_CHECK=1")
+
+	versionPre := buildinfo.Version().Pre
+
+	if versionPre != nil {
```

Comment:
probably need to check len instead of nil, because an empty list can be non-nil in go.

```go
l := make([]string, 0)
fmt.Println(l == nil)
// false
```

but a nil list will return len 0:

```go
var n []string
fmt.Println(n == nil)
// true
fmt.Println(len(n) == 0)
// true
```

Reply:
```suggestion
	if len(versionPre) > 0 {
```

Reply:
Thank you, I wasnt aware of that!